static/js/NavSidebar.jsx

```diff
@@ -85,6 +87,59 @@ const TitledText = ({enTitle, heTitle, enText, heText}) => {
   </Module>
 };
 
+const RecentlyViewed = ({toggleSignUpModal, mobile}) => {
+   const [recentlyViewedItems, setRecentlyViewedItems] = useState(null);
+   const handleAllHistory = (e) => {
+    if (!Sefaria._uid) {
+      e.preventDefault();
+      toggleSignUpModal(SignUpModalKind.ViewHistory);
+    }
+    gtag('event', 'recently_viewed', {link_type: 'all_history', logged_in: !!Sefaria._uid});
+   }
+   const trackItem = (ref, pos) => {
+     gtag('event', 'recently_viewed', {link_text: ref, link_type: 'ref', link_pos: pos})
+   }
+   const filterRecentlyViewedItems = () => {
+        let itemsToShow = [];
+        let booksFound = [];
+        Sefaria.userHistory.items.forEach(x => {
+        if (!booksFound.includes(x.book) && x.book !== "Sheet") {
+           booksFound.push(x.book);
+           itemsToShow.push(x);
+        }});
+
+        itemsToShow = itemsToShow.slice(0, 3).map((x, i) => {
+            return <li><a href={x.ref} onClick={() => trackItem(x.ref, i+1)}>{Sefaria.interfaceLang === "hebrew" ? x.he_ref : x.ref}</a></li>;
+         });
+        setRecentlyViewedItems(itemsToShow);
```

Comment:
I'd rather you set state with data, not html. then use said state later on in the render process. 
also, if its being repeated it might be wroth createing a small component to encapsulate the render of the item

Reply:
In my most recent push, I fixed this by setting state with data rather than HTML.  I created a small component RecentlyViewedList as well so as not to duplicate code.